src/Data/Bool/Properties.agda

```diff
@@ -626,8 +626,9 @@ T-irrelevant : U.Irrelevant T
 T-irrelevant {true}  _  _  = refl
 
 T? : U.Decidable T
-T? true  = yes _
-T? false = no (λ ())
+isYes (T? x) = x
+reflects (T? false) = false id
+reflects (T? true ) = true  _
 
```

Comment:
Could we have a smart constructor for this kind of pattern?
Something like (we probably need a different name):
`if_then_else_ : (b : Bool) -> P true -> P false -> Dec P`